### DISCUSSION TITLE
Yjs bindings similar to valtio-yjs

### DISCUSSION THREAD

**Discussion by raymonddaikon** [LegendApp/legend-state#172] [Ideas]
I was wondering if there were any plans on integrating Yjs bindings similar to valtio-yjs?

**Comment by jmeistrich** [comment]
We don't have any plans specifically, but it doesn't be too difficult. I imagine it could look pretty similar to valtio-yjs or another implementation? Would you like to try submitting a PR for it?

**Reply by raymonddaikon** [reply]
Yes I'll open up a pr once I get my implementation working

**Comment by andrictham** [comment]
@raymonddaikon Interested in how you implemented Yjs with Legend State. What text editor and backend provider did you go with?

I’m trying to figure out how to get Yjs & Tiptap to play nice with Legend State at the moment and still deciding on a backend provider (Hocuspocus, Liveblocks, or PartyKit).

I have some state outside of TipTap that’s in Legend, that I want to keep in the same Yjs doc so it all syncs up.

**Reply by raymonddaikon** [reply]
Here's a gist of what I did, its 90% of the way there, I essentially copied valtio-yjs but changed it to use legend state getters and setters. https://gist.github.com/raymonddaikon/af4e808656b3df54a5b7d0aa90d81d95
You'll need to refactor the `subscribeProxy` as I manually handle transactions. But you'll have to transform legendstate proxy values into Yjs values.

**Reply by jmeistrich** [reply]
@raymonddaikon Would you like to clean up that last 10% (and ideally add a few tests) and we could look into making it an official plugin?
